custom_components/tuya_local/devices/calex_convector_heater.yaml

```diff
@@ -0,0 +1,159 @@
+      type: boolean
+      mapping:
+        - dps_val: true
+          value: "heat"
+          icon: "mdi:radiator"
+        - dps_val: false
+          value: "off"
+          icon: "mdi:radiator-disabled"
+    - id: 2
+      name: temperature
+      type: integer
+      unit: ℃
+      range:
+        min: 5
+        max: 35
+    - id: 3
+      name: current_temperature
+      readonly: true
+      type: integer
+      unit: ℃
+      range:
+        min: -20
+        max: 50
+    - id: 101
+      name: preset_mode
+      type: string
+      mapping:
+        - dps_val: H1
+          value: eco
+        - dps_val: H2
+          value: boost
+        - dps_val: AF
+          value: away
+        - dps_val: P7
+          value: activity
+secondary_entities:
+  - entity: lock
+    name: Child lock
+    category: config
+    dps:
+      - id: 7
+        type: boolean
+        name: lock
+        mapping:
+          - dps_val: true
+            icon: "mdi:hand-back-right-off"
+          - dps_val: false
+            icon: "mdi:hand-back-right"
+  - entity: sensor
+    name: Status
+    class: enum
+    category: diagnostic
+    dps:
+      - id: 11
+        name: sensor
+        type: string
+        mapping:
+          - dps_val: standby
+            value: standby
+            icon: "mdi:fire-off"
+          - dps_val: working
+            value: working
+            icon: "mdi:fire"
+  - entity: select
+    name: Timer
+    icon: "mdi:timer"
+    category: config
+    dps:
+      - id: 19
+        type: string
+        name: option
+        mapping:
+          - dps_val: cancel
+            value: "Off"
+          - dps_val: 1h
+            value: 1 hour
+          - dps_val: 2h
+            value: 2 hours
+          - dps_val: 3h
+            value: 3 hours
+          - dps_val: 4h
+            value: 4 hours
+          - dps_val: 5h
+            value: 5 hours
+          - dps_val: 6h
+            value: 6 hours
+          - dps_val: 7h
+            value: 7 hours
+          - dps_val: 8h
+            value: 8 hours
+          - dps_val: 9h
+            value: 9 hours
+          - dps_val: 10h
+            value: 10 hours
+          - dps_val: 11h
+            value: 11 hours
+          - dps_val: 12h
+            value: 12 hours
+          - dps_val: 13h
+            value: 13 hours
+          - dps_val: 14h
+            value: 14 hours
+          - dps_val: 15h
+            value: 15 hours
+          - dps_val: 16h
+            value: 16 hours
+          - dps_val: 17h
+            value: 17 hours
+          - dps_val: 18h
+            value: 18 hours
+          - dps_val: 19h
+            value: 19 hours
+          - dps_val: 20h
+            value: 20 hours
+          - dps_val: 21h
+            value: 21 hours
+          - dps_val: 22h
+            value: 22 hours
+          - dps_val: 23h
+            value: 23 hours
+          - dps_val: 24h
+            value: 24 hours
+  - entity: sensor
+    name: Timer left, min
```

Comment:
Put the "min" as the unit instead of the name, add a class of duration.  I also prefer "Time remaining" rather than left, as it doesn't have other meanings so is clearer (left/right).

Reply:
ack